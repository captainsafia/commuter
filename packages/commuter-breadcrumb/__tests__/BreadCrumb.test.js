import React from 'react';
import renderer from 'react-test-renderer';

import BreadCrumb from '../src';

it('renders correctly', () => {
  const tree = renderer.create(
    <BreadCrumb
      path={"path/for/tests"}
      onClick={() => {}}
      basepath={"/view"}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot(); 
});

it('renders the correct number of elements', () => {
  const tree = renderer.create(
    <BreadCrumb
      path={"path/for/tests"}
      onClick={() => {}}
      basepath={"/view"}
    />
  ).toJSON();
  expect(tree.children.length).toEqual(7);
});
