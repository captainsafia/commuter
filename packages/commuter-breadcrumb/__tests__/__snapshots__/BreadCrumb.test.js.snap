// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui breadcrumb"
>
  <div
    className="section"
    href={undefined}
    onClick={[Function]}
  >
    <a
      href="/view/"
      onClick={[Function]}
    >
      Home
    </a>
  </div>
  <i
    aria-hidden="true"
    className="right angle icon divider"
  />
  <div
    className="section"
    href={undefined}
    onClick={[Function]}
  >
    <a
      href="/view/path/"
      onClick={[Function]}
    >
      path
    </a>
  </div>
  <i
    aria-hidden="true"
    className="right angle icon divider"
  />
  <div
    className="section"
    href={undefined}
    onClick={[Function]}
  >
    <a
      href="/view/path/for/"
      onClick={[Function]}
    >
      for
    </a>
  </div>
  <i
    aria-hidden="true"
    className="right angle icon divider"
  />
  <div
    className="active section"
    href={undefined}
    onClick={[Function]}
  >
    tests
  </div>
</div>
`;
